* Emacs runs debian-startup and sets debian-emacs-flavor.
  Patch: debian-startup.diff

  * Emacs runs debian-startup during the startup process unless
    site-run-file is false.
    
  * The global variable debian-emacs-flavor is bound to 'emacs22.
    Author: Rob Browning <rlb@defaultvalue.org>
    
Index: sid/lisp/startup.el
===================================================================
--- sid.orig/lisp/startup.el
+++ sid/lisp/startup.el
@@ -302,6 +302,10 @@
 Warning Warning!!!  Pure space overflow    !!!Warning Warning
 \(See the node Pure Storage in the Lisp manual for details.)\n")
 
+(defconst debian-emacs-flavor 'emacs22
+  "A symbol representing the particular debian flavor of emacs running.
+Something like 'emacs20, 'xemacs20, etc.")
+
 (defun normal-top-level-add-subdirs-to-load-path ()
   "Add all subdirectories of current directory to `load-path'.
 More precisely, this uses only the subdirectories whose names
@@ -826,8 +830,21 @@
     ;; be loaded from site-run-file and wants to test if -q was given
     ;; should check init-file-user instead, since that is already set.
     ;; See cus-edit.el for an example.
-    (if site-run-file
-	(load site-run-file t t))
+
+    ;; Original upstream startup
+    ;; (if site-run-file
+    ;;     (load site-run-file t t))
+    ;;
+
+    ;; Debian startup
+     (if site-run-file
+        (progn
+          ;; Load all the debian package snippets.
+          ;; It's in here because we want -q to kill it too.
+          (if (load "debian-startup" t t nil)
+              (debian-startup debian-emacs-flavor))
+          ;; Now the normal site file...
+          (load site-run-file t t nil)))
 
     ;; Sites should not disable this.  Only individuals should disable
     ;; the startup screen.
